Rubberduck.Inspections/Concrete/EmptyBlockInspectionListenerBase.cs

```diff
@@ -0,0 +1,81 @@
+﻿using Antlr4.Runtime;
+using Rubberduck.Parsing;
+using Rubberduck.Parsing.Grammar;
+using Rubberduck.Parsing.Inspections.Abstract;
+using Rubberduck.VBEditor;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Rubberduck.Inspections.Concrete
+{
+    public class EmptyBlockInspectionListenerBase : VBAParserBaseListener, IInspectionListener
+    {
+        private readonly List<QualifiedContext<ParserRuleContext>> _contexts = new List<QualifiedContext<ParserRuleContext>>();
+        public IReadOnlyList<QualifiedContext<ParserRuleContext>> Contexts => _contexts;
+
+        public QualifiedModuleName CurrentModuleName { get; set; }
+
+        public void ClearContexts()
+        {
+            _contexts.Clear();
+        }
+
+        public void InspectBlockForExecutableStatements<T>(VBAParser.BlockContext block, T context) where T : ParserRuleContext
+        {
+            if (!BlockContainsExecutableStatements(block))
+            {
+                AddResult(new QualifiedContext<ParserRuleContext>(CurrentModuleName, context));
+            }
+        }
+
+        public void AddResult(QualifiedContext<ParserRuleContext> qualifiedContext)
+        {
+            _contexts.Add(qualifiedContext);
+        }
+
+        private bool BlockContainsExecutableStatements(VBAParser.BlockContext block)
+        {
+            return block != null && block.children != null && ContainsExecutableStatements(block);
+        }
+
+        private bool ContainsExecutableStatements(VBAParser.BlockContext block)
+        {
+            foreach (var child in block.children)
+            {
+                if (child is VBAParser.BlockStmtContext)
```

Comment:
could consider a recursive call?

Reply:
@Vogel612 no need, the method works by process of elimination; only known non-statements keep the loop going, anything else (including a loop or `If` statement) `return true` runs and we're already out.